subprojects/shipkit/src/main/groovy/org/shipkit/internal/gradle/java/tasks/CreateDependencyInfoFile.java

```diff
@@ -0,0 +1,76 @@
+package org.shipkit.internal.gradle.java.tasks;
+
+import org.gradle.api.artifacts.*;
+import org.json.simple.JsonArray;
+import org.json.simple.JsonObject;
+import org.json.simple.Jsoner;
+import org.shipkit.internal.notes.util.IOUtil;
+
+import java.io.File;
+import java.util.ArrayList;
+import java.util.LinkedHashMap;
+import java.util.Set;
+
+public class CreateDependencyInfoFile {
+
+    private static final String DESCRIPTION = "This file was generated by Shipkit Gradle plugin. See http://shipkit.org\n\n";
+
+    public void createDependencyInfoFile(CreateDependencyInfoFileTask task) {
+        JsonObject result = createJsonObject();
+        result.put("description", DESCRIPTION);
+
+        JsonArray dependencies = createJsonArray();
+
+        for (Dependency dependency: task.getConfiguration().getAllDependencies()) {
```

Comment:
I think it is a good idea to work with declared dependencies rather than resolved configuration. I have tested it. I configured dependency resolution rules that force different version. In result, the effective resolved dependency graph had different versions than declared versions. The pom file was still based on declared versions.

Nice catch!